daemon/daemon.go

```diff
@@ -1139,7 +1139,14 @@ func NewDaemon() (*Daemon, error) {
 
 	// Set up ipam conf after init() because we might be running d.conf.KVStoreIPv4Registration
 	log.Info("Initializing IPAM")
-	switch err := ipam.Init(); err.(type) {
+	ipam.Init()
+
+	restoredEndpoints, err := d.restoreOldEndpoints(option.Config.StateDir, true)
```

Comment:
Could you add a comment indicating endpoints are being restored before internal IPs are allocated?

Reply:
Fixed